### PR TITLE
Fixes #3630 - ClientContext updated to handle time duration values correctly

### LINKED ISSUE #3630: ClientContext performing incorrect conversion of PropertyType.TIMEDURATION values
**Describe the bug**

_ClientContext.getBatchWriterConfig(props)_ is incorrectly handling [BATCH_WRITER_LATENCY_MAX ](https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java#L370) and [BATCH_WRITER_TIMEOUT_MAX](https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java#L370), such that the configured values are being converted from seconds to milliseconds twice, causing the stored values to be off by a factor of 1000

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Accumulo version: 2.1

**To Reproduce**
Use _ClientContext.getBatchWriterConfig(props)_ to build a BatchWriterConfig with the following properties
```
batch.writer.timeout.max=300s
batch.writer.latency.max=100s
```
And observe that the resulting BWC's values are incorrect:
```
[maxMemory=52428800, maxLatency=100000000, maxWriteThreads=3, timeout=300000000, durability=DEFAULT]
```

**Expected behavior**

The stored values for the BWC instance should be:
- maxLatency = 100000
- timeout =  300000

### PR DESCRIPTION
**Pull request by keith-ratcliffe**

### CODE REVIEW

File: core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java [hunk 1]
```diff
@@ -159,10 +159,9 @@ public void testGetBatchWriterConfigNotUsingDefaults() {
         .getMemoryAsBytes(ClientProperty.BATCH_WRITER_MEMORY_MAX.getValue(props));
     assertEquals(expectedMemory, batchWriterConfig.getMaxMemory());
 
-    assertEquals(Long.MAX_VALUE, batchWriterConfig.getMaxLatency(MILLISECONDS));
+    assertEquals(40, batchWriterConfig.getMaxLatency(SECONDS));
```
Comment (dlmarion) [inline core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java:162]: Shouldn't the units here be milliseconds?

File: core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java [hunk 2]
```diff
@@ -159,10 +159,9 @@ public void testGetBatchWriterConfigNotUsingDefaults() {
...
 
-    // getTimeout returns time in milliseconds, therefore the 15 becomes 15000.
-    assertEquals(15000, batchWriterConfig.getTimeout(SECONDS));
+    assertEquals(15, batchWriterConfig.getTimeout(SECONDS));
```
Comment (dlmarion) [inline core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java:164]: Shouldn't the units here be milliseconds?

File: core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java (hunk 1 above)
Comment (keith-ratcliffe) [reply core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java:162]: Sorry, I may be misunderstanding the question. The TimeUnit arg to getMaxLatency could be changed to MILLISECONDS, assuming a corresponding change in _expected_ arg from 40 to 40000. But that wouldn't change the validity of the assertion, right?

File: core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java (hunk 2 above)
Comment (keith-ratcliffe) [reply core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java:164]: Same reply as above

**Comment by keith-ratcliffe** [comment]
Same bug [here](https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java#L436) which I missed, for conditional writer timeout. The test uses an prop value of "17" (defaulting to time unit of SECONDS) and then [asserts that 17000 seconds is expected](https://github.com/apache/accumulo/blob/2.1/core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java#L219)

**Comment by keith-ratcliffe** [comment]
Rebased/squash commit now includes the fix for CONDITIONAL_WRITER_TIMEOUT_MAX and also cleanup/refinement of the affected junits in ClientContextTest

File: core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java (hunk 1 above)
Comment (dlmarion) [reply core/src/test/java/org/apache/accumulo/core/clientImpl/ClientContextTest.java:157]: I see, the default without units is SECONDS.

**Comment by dlmarion** [comment]
@keith-ratcliffe - Thanks for your contribution. I see this is your first time contributing. Feel free to put up a PR on https://github.com/apache/accumulo-website to add yourself to the people page.

**Pull request merged**